### PR TITLE
Submissions v4 APIs

### PR DESCRIPTION
**Pull request by paul-burnett**

### CODE REVIEW

File: src/connectors/registrationsDb-v4/registrationsDb.v4.connector.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+
     const registrations = await registrationsCollection
       .find(
         {
           $and: [
             { "fsa-rn": { $not: { $regex: /^tmp_/ } } },
-            { local_council_url: council },
+            { local_council_url: { $in: councils } },
             { collected: { $in: collected } },
             { reg_submission_date: { $gte: new Date(after) } },
```
[the remark this answers is not in the document]
Comment (simon-says-test) [reply src/connectors/registrationsDb-v4/registrationsDb.v4.connector.js:19]: in fact there's a fair bit of that

[the remark this answers is not in the document]
Comment (paul-burnett) [reply src/connectors/registrationsDb-v4/registrationsDb.v4.connector.js]: Removed some of the duplication, routers and controllers still there which enables easier removal of old API versions in future.

File: tests/collections/e2e/getSingleRegistration.v4.api.e2e.test.js [hunk 2]
```diff
@@ -46,11 +49,58 @@ describe("Get single registration through API", () => {
       expect(response.establishment).toBeDefined();
```
[the remark this answers is not in the document]
Comment (paul-burnett) [reply tests/collections/e2e/getSingleRegistration.v4.api.e2e.test.js:49]: I think these tests don't guarantee what data is in the DB, they just check the response is as expected.
New fields in response are verified in other tests.

File: tests/submissions/e2e/postDirectRegistrationSubmission.v4.api.e2e.test.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    it("should successfully submit the registration and return a fsa-rn", () => {
+      expect(postResponse["fsa-rn"]).toBeDefined();
+    });
+    it("should be retrievable through the collcetions API", () => {
+      expect(getResponse.fsa_rn).toBe(postResponse["fsa-rn"]);
+      expect(getResponse.establishment).toBeDefined();
+      expect(getResponse.establishment.operator.operator_first_name).toBe(
+        registration.establishment.operator.operator_first_name
+      );
+      expect(getResponse.establishment.activities.water_supply).toBe(
```
[the remark this answers is not in the document]
Comment (paul-burnett) [reply tests/submissions/e2e/postDirectRegistrationSubmission.v4.api.e2e.test.js:107]: Good point, added checks for the new fields

File: tests/collections/component/getAllRegistrationsByCouncil.v4.component.test.js [hunk 4]
```diff
@@ -9,18 +9,21 @@ const { logEmitter } = require("../../../src/services/logging.service");
...
 
+jest.setTimeout(30000);
+
 const frontendSubmitRegistration = async () => {
   try {
     for (let index in mockRegistrationData) {
       const requestOptions = {
         method: "POST",
         data: mockRegistrationData[index],
+        body: JSON.stringify(mockRegistrationData[index]),
```
[the remark this answers is not in the document]
Comment (paul-burnett) [reply tests/collections/component/getAllRegistrationsByCouncil.v4.component.test.js]: Removed, thanks

**Review by simon-says-test — APPROVED** [review]

**Pull request merged**